api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubV3.java

```
@@ -0,0 +1,196 @@
+
+    protected CollectorItem buildCollectorItem (ObjectId collectorId, String repoUrl, String branch) {
+        if (!StringUtils.isEmpty(repoUrl) && !StringUtils.isEmpty(branch)) {
+            CollectorItem collectorItem = new CollectorItem();
+            collectorItem.setCollectorId(collectorId);
+            collectorItem.setEnabled(true);
+            collectorItem.setPushed(true);
+            collectorItem.setLastUpdated(System.currentTimeMillis());
+            collectorItem.getOptions().put(REPO_URL, repoUrl);
+            collectorItem.getOptions().put(BRANCH, branch);
+
+            return collectorItem;
+        }
+
+        return null;
+    }
+
+    protected CollectorItem getCollectorItem(String repoUrl, String branch) throws HygieiaException {
+        Collector col = collectorService.createCollector(getCollector());
+
+        if (col == null)
+            throw new HygieiaException("Failed creating collector.", HygieiaException.COLLECTOR_CREATE_ERROR);
+
+        CollectorItem item = buildCollectorItem(col.getId(), repoUrl, branch);
+        if (item == null)
+            throw new HygieiaException("Failed creating collector item. Invalid repo url and/or branch", HygieiaException.COLLECTOR_ITEM_CREATE_ERROR);
+
+        CollectorItem colItem = collectorService.createCollectorItem(item);
+
+        if (colItem == null)
+            throw new HygieiaException("Failed creating collector item.", HygieiaException.COLLECTOR_ITEM_CREATE_ERROR);
+
+        return colItem;
+    }
+
+    protected String getRepositoryToken(String scmUrl) {
+        Collector collector = collectorService.createCollector(getCollector());
+
+        Map<String, Object> options = new HashMap<>();
+        options.put(REPO_URL, scmUrl);
+
+        GitHubRepo gitHubRepo = findByCollectorIdAndOptions(collector.getId(), options);
+
+        if (gitHubRepo != null) {
+            return String.valueOf(gitHubRepo.getOptions().get(TOKEN));
```

if () return null; fail fast.

---

Have changed this, thanks.